src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -47,73 +63,709 @@ static ColliderJntSphInit D_80A17040 = {
+        EnFish_Respawning_SetupFleePlayer(this);
+    }
+}
+
+// The three following actionfunctions also turn the yaw to home if the fish is too far from it.
+
+void EnFish_Respawning_SetupFollowChild(EnFish* this) {
+    this->actor.gravity = 0.0f;
+    this->actor.minVelocityY = 0.0f;
+    this->timer = Rand_S16Offset(15, 45);
+    this->unk_250 = 0;
+    EnFish_SetInWaterAnimation(this);
+    this->actionFunc = EnFish_Respawning_FollowChild;
+}
+
+void EnFish_Respawning_FollowChild(EnFish* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    EnFish_SetYOffset(this);
+    Math_SmoothStepToF(&this->actor.speedXZ, 1.8f, 0.08f, 0.4f, 0.0f);
+
+    if ((EnFish_XZDistanceSquared(&this->actor.world.pos, &this->actor.home.pos) > SQ(80.0f)) || (this->timer < 4)) {
+        Math_StepToAngleS(&this->actor.world.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.home.pos),
+                          3000);
+    } else {
+        if ((this->actor.child != NULL) && (&this->actor != this->actor.child)) {
+            Math_StepToAngleS(&this->actor.world.rot.y,
+                              Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.child->world.pos), 3000);
+        }
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->skelAnime.playSpeed = CLAMP_MAX((this->actor.speedXZ * 1.5f) + 0.8f, 4.0f);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer <= 0) {
+        EnFish_Respawning_SetupSlowDown(this);
+    } else if (&this->actor == this->actor.child) {
+        EnFish_Respawning_SetupApproachPlayer(this);
+    } else if (EnFish_CheckXZDistanceToPlayer(this, globalCtx)) {
+        EnFish_Respawning_SetupFleePlayer(this);
+    }
+}
+
+void EnFish_Respawning_SetupFleePlayer(EnFish* this) {
+    this->actor.gravity = 0.0f;
+    this->actor.minVelocityY = 0.0f;
+    this->timer = Rand_S16Offset(10, 40);
+    this->unk_250 = 0;
+    EnFish_SetInWaterAnimation(this);
+    this->actionFunc = EnFish_Respawning_FleePlayer;
+}
+
+void EnFish_Respawning_FleePlayer(EnFish* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 pad2;
+    s16 frames;
+    s16 yaw;
+    s16 playerClose;
+
+    EnFish_SetYOffset(this);
+    playerClose = EnFish_CheckXZDistanceToPlayer(this, globalCtx);
+    Math_SmoothStepToF(&this->actor.speedXZ, 4.2f, 0.08f, 1.4f, 0.0f);
+
+    if (EnFish_XZDistanceSquared(&this->actor.world.pos, &this->actor.home.pos) > SQ(160.0f)) {
+        yaw = Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.home.pos);
+        Math_StepToAngleS(&this->actor.world.rot.y, yaw, 3000);
+    } else if ((this->actor.child != NULL) && (&this->actor != this->actor.child)) {
+        yaw = Math_Vec3f_Yaw(&this->actor.world.pos, &this->actor.child->world.pos);
+        Math_StepToAngleS(&this->actor.world.rot.y, yaw, 2000);
+    } else if (playerClose) {
+        yaw = this->actor.yawTowardsPlayer + 0x8000;
+        frames = globalCtx->state.frames;
+
+        if (frames & 0x10) {
```

Comment:
Does `%` not work here?

Reply:
No, they all have to be `&`, unfortunately.